ouroboros-network/src/Ouroboros/Network/PeerSelection/LedgerPeers.hs

```diff
@@ -90,11 +90,17 @@ getLedgerPeers (LedgerPeersConsensusInterface lpGetLatestSlot
                                               lpGetLedgerStateJudgement
                                               lpGetLedgerPeers)
                slot = do
-  curSlot <- lpGetLatestSlot
-  if curSlot < slot
-     then pure BeforeSlot
-     else LedgerPeers <$> lpGetLedgerStateJudgement
-                      <*> lpGetLedgerPeers
+  wOrigin <- lpGetLatestSlot
+  case wOrigin of
+    Origin | slot == 0          -> buildLedgerPeers
```

Comment:
It would be more consistent if we use the same convention as `consensus`, and use `slot :: WithOrigin SlotNo`  instead of `slot :: SlotNo`.  This will require small modifications up the stack up to topology parser were we'd interpret `0` as `Origin`.

This could be done later though, if you prefer that way please create an issue.

Reply:
I think we can actually use `UseLedgerPeers` value instead